src/migration/1714633817175-SquadTotalPostsTrigger.ts

```diff
@@ -0,0 +1,65 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class SquadTotalPostsTrigger1714633817175 implements MigrationInterface {
+  name = 'SquadTotalPostsTrigger1714633817175';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION increment_squad_posts_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE  source
+          SET     flags = jsonb_set(flags, '{totalPosts}', to_jsonb(COALESCE(CAST(flags->>'totalPosts' AS INTEGER) , 0) + 1))
+          WHERE   id = NEW."sourceId"
+          AND     type = 'squad';
+          RETURN NEW;
+        END;
+        $$
+      `);
+    queryRunner.query(
+      `CREATE TRIGGER increment_squad_posts_count AFTER INSERT ON "post" FOR EACH ROW EXECUTE PROCEDURE increment_squad_posts_count()`,
+    );
+
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION deduct_squad_stats()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          IF OLD.deleted = false AND NEW.deleted = true THEN
+            UPDATE  source
+            SET     flags = jsonb_set(
+                              jsonb_set(
+                                flags,
+                                '{totalPosts}',
+                                to_jsonb(GREATEST(0, CAST(flags->>'totalPosts' AS INTEGER) - 1))
+                              ),
+                              '{totalViews}',
+                              to_jsonb(GREATEST(0, CAST(flags->>'totalViews' AS INTEGER) - OLD.views))
```

Comment:
Also not sure if we need to decrement views when post is deleted, views are still there and happened, no matter if post is deleted.

Reply:
Sorry, asked in discussion, did not see it before https://dailydotdev.slack.com/archives/C069XNYMDGV/p1714669514654279

Reply:
Resolved in https://github.com/dailydotdev/daily-api/pull/1866#discussion_r1589028400

Reply:
Yeah, sounds good to me. That was also the reason I asked it, I wanted the deduction to not happen. Let's do it.